lambda scripts/build_api.sh

```diff
@@ -26,57 +26,21 @@ if [ "$BASEDIR" != "bin" ]; then
 fi
 
 
-## FUNCTIONS
-## -----------------------------------------------------------------------------
-function check_rc () {
-  ## Checks the return code of call and if not equal to 0, emits an error and
-  ## exits the script with a failure.
-  ##
-  ## Args:
-  ##   $1 - Return Code from command
-  ##   $2 - Error message if failure occurs.
-
-  let RC=$1
-  MESSAGE=$2
-
-  if [ $RC -ne 0 ]; then
-      >&2 echo "$MESSAGE"
-      deactivate
-      exit 1
-  fi
-}
+source ../../bin/common/check_returncode.sh
+source ../../bin/common/venv_management.sh
 
 
 ## MAIN
 ## -----------------------------------------------------------------------------
-## Create the virtual env. Remove it if it already exists.
-echo "INFO: Creating virtual environment ..."
-if [ -d venv ]; then
-  rm -rf venv
-  find . -type d -name "__pycache__" -exec rm -rf {} +
-fi
-
-python3 -m venv venv
-source venv/bin/activate
+run_and_check_returncode "create_and_activate_venv"
+trap 'deactivate_and_delete_venv;rm -rf build;' EXIT
```

Comment:
Copy paste error here. should just be `trap 'deactivate_and_delete_venv'` since there is no build directory created in this script.

Reply:
Fixed